### PR TITLE
Make sure clusterSpecifier always gets set during translation loop

### PR DESCRIPTION
**Pull request by jenshu**
This helps avoid potential NPEs.
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/5112

### CODE REVIEW

File: projects/gloo/pkg/translator/route_config.go [hunk 1]
```diff
@@ -358,13 +358,12 @@ func (t *translatorInstance) setAction(
...
 	case *v1.RouteAction_Single:
+		out.ClusterSpecifier = &envoy_config_route_v3.RouteAction_Cluster{}
 		usRef, err := usconversion.DestinationToUpstreamRef(dest.Single)
 		if err != nil {
 			return err
 		}
-		out.ClusterSpecifier = &envoy_config_route_v3.RouteAction_Cluster{
-			Cluster: UpstreamToClusterName(usRef),
-		}
+		out.ClusterSpecifier.(*envoy_config_route_v3.RouteAction_Cluster).Cluster = UpstreamToClusterName(usRef)
```
Comment (MitchAman) [inline projects/gloo/pkg/translator/route_config.go:366]:
```suggestion
		out.GetClusterSpecifier().(*envoy_config_route_v3.RouteAction_Cluster).Cluster = UpstreamToClusterName(usRef)
```

Comment (jenshu) [reply projects/gloo/pkg/translator/route_config.go:366]: i'm not sure this buys us much, if `out` was nil it would have errored already on line 361

Comment (MitchAman) [reply projects/gloo/pkg/translator/route_config.go:366]: I agree, but for whatever reason codegen wants it to look that way

Comment (jenshu) [reply projects/gloo/pkg/translator/route_config.go:366]: ah good call, will fix

**Review by MitchAman — APPROVED** [review]

**Pull request merged**